clusterman/draining/queue.py

```diff
@@ -373,29 +376,45 @@ def clean_processing_hosts_cache(self) -> None:
         host_to_process = self.get_warned_host()
         if host_to_process:
             logger.info(f"Processing spot warning for {host_to_process.hostname}")
-            spot_fleet_resource_groups = []
-            for pool in get_pool_name_list(self.cluster, "mesos"):  # draining only supported for Mesos clusters
+            spot_fleet_resource_groups: Set[str] = set()
+            autoscaling_resource_groups: Set[str] = set()
+            # we do this in two loops since we only use SFRs for Mesos, but we use Spot
+            # ASGs for Kubernetes
+            for pool in get_pool_name_list(self.cluster, "mesos"):
                 pool_config = staticconf.NamespaceReaders(POOL_NAMESPACE.format(pool=pool, scheduler="mesos"))
                 for resource_group_conf in pool_config.read_list("resource_groups"):
-                    spot_fleet_resource_groups.extend(
-                        list(
-                            SpotFleetResourceGroup.load(
-                                cluster=self.cluster,
-                                pool=pool,
-                                config=list(resource_group_conf.values())[0],
-                            ).keys()
-                        )
+                    spot_fleet_resource_groups |= set(
+                        SpotFleetResourceGroup.load(
+                            cluster=self.cluster,
+                            pool=pool,
+                            config=list(resource_group_conf.values())[0],
+                        ).keys()
+                    )
+
+            for pool in get_pool_name_list(self.cluster, "kubernetes"):
+                pool_config = staticconf.NamespaceReaders(POOL_NAMESPACE.format(pool=pool, scheduler="kubernetes"))
+                for resource_group_conf in pool_config.read_list("resource_groups"):
+                    autoscaling_resource_groups |= set(
+                        AutoScalingResourceGroup.load(
+                            cluster=self.cluster,
+                            pool=pool,
+                            config=list(resource_group_conf.values())[0],
+                        ).keys()
                     )
 
             # we should definitely ignore termination warnings that aren't from this
             # cluster or maybe not even paasta instances...
             if host_to_process.group_id in spot_fleet_resource_groups:
-                logger.info(f"Sending spot warned host to drain: {host_to_process.hostname}")
+                logger.info(f"Sending spot fleet warned host to drain: {host_to_process.hostname}")
                 self.submit_host_for_draining(host_to_process)
+            if host_to_process.group_id in autoscaling_resource_groups:
+                logger.info(f"Cleaning-up and sending spot warned host to terminate: {host_to_process.hostname}")
+                k8s_clean_node(kube_operator_client, host_to_process.agent_id)
+                self.submit_host_for_termination(host_to_process, 0)
```

Comment:
suggestion:
```suggestion
                self.submit_host_for_termination(host_to_process, delay=0)
```
makes this line clearer at a glance